torch_geometric/nn/conv/message_passing.py

```diff
@@ -197,10 +197,11 @@ def __collect__(self, args, edge_index, size, kwargs):
 
         return out
 
-    def propagate(self, edge_index: Adj, size: Size = None, **kwargs):
+    def propagate(self, edge_index: Adj, size: Size = None, decomposed_layer = 1, **kwargs):
```

Comment:
I think the `decomposed_layer` argument should be passed to the `__init__` function in the message passing class. If its an argument to `propogate` then users will have to modify the call to `propogate` in the `forward` of conv layers like `GCNConv` to use feature decomposition. But if its an argument to `__init__` of message passing then the feature decomposition can be used as `GCNConv(16, 32, decomposed_layer =2)`.